src/main/java/gov/nasa/pds/harvest/cfg/model/AutogenCfg.java

```diff
@@ -13,4 +13,5 @@
     public Set<String> classFilterIncludes = new HashSet<>();
     public Set<String> classFilterExcludes = new HashSet<>();
     public Set<String> dateFields = new HashSet<>();
+    public boolean generate = true;
```

Comment:
Was there specific reason to directly use public variables here? It is generally not recommended to have public variables in Java classes due the to reasons mentioned in the following Sonar rule.

**Class variable fields should not have public accessibility**
https://rules.sonarsource.com/java/RSPEC-1104

It is recommended to use private attributes and public accessor methods (set and get).

Reply:
Yes. This class is an equivalent of a structure in other programming languages like C/C++. There is no point to create getters and setters.

Reply:
I understand the point made by @tdddblog.  

@tloubrieu-jpl, @jordanpadams and @nutjob4life, I am asking this, since I am new to the project. Do we have a list of coding guidelines on debatable areas like this? For example, do we have any guideline saying we **must always** follow the OOP principle of encapsulation.

Reply:
@ramesh-maddegoda we do not. [JPL has one](http://lars.jpl.nasa.gov/jpl-java-standard.pdf) but it doesn't look like the link is working. that being said, that is more of a guide, versus a requirement. we have not encountered any debate that couldn't be worked out in a breakout, but I think we can negotiate those when the time comes. we can discuss this one in more detail at the breakout.

that being said, regardless, thanks for the review! glad you dove into the weeds here.